pkg/controller/master/storagenetwork/storage_network.go

```diff
@@ -397,6 +407,58 @@ func (h *Handler) removeOldNad(setting *harvesterv1.Setting) error {
 	return nil
 }
 
+// check if LH Pod backing-image,backing-image-ds, instance-manager status after configuring storage network
+func (h *Handler) checkLHPodStatus() error {
```

Comment:
It seems not the best idea to check LH related PODs, which will enhance the coupling.

The SN is using Whereabouts to allocate IPs to LH PODs, could we check it, if none IP is allocated after certain time;  and if IPs are exhausted, add related condition/status as well.

Reply:
@w13915984028 Thanks for reviewing.
Can you please elaborate on "enhance the coupling" problem ?
when you say "check none IP allocated after sometime", is it to check on the pod or is it some other way ? 

Currently whereabouts does the allocatable ip addresses internally which is not known to the user unless it tries that on a pod and pod fails.

Reply:
one thing I notice about wherabout ippools is, it does have only allocations list (not any other msg regarding failures), so there could be cases where it would have allocated partially, so getting a node count and if backing-image-ds is present,will make sure number of IPs required in the cluster and this can be compared to the number of ips allocated by whereabouts ippool to know if all IPs are allocated successfully.

Pros:
1.Use whereabouts ip allocation status instead of longhorn pod status.
2.IP count needed could be derived from above and validated against number of IPs available from the range given by user instead of validating against some minimum value.

Cons:
1.Makes the code little complex.

Reply:
Yeah, this does sound a bit complicated and potentially brittle. Overall, is it the storage network controller's responsibility to watch for LH pods' health? The pods can fail for different reasons, potentially, not even related to storage network change. Ideally, degrading pods' health should be reported by the monitoring system, not a `setting` controller.

I think the webhook validation makes sense, but not sure about the LH pods' health and UI reporting part. Is it common for us to report errors in the setting UI? From an UX perspective, it's likely that the errors won't show up right away, right? If user navigated away from the settings page, they won't see any later errors.